pike/model.py

```diff
@@ -2020,23 +2041,17 @@ def __init__(self, tree, smb_res, create_guid=None, prev=None):
 
         if self.oplock_level != smb2.SMB2_OPLOCK_LEVEL_NONE:
             if self.oplock_level == smb2.SMB2_OPLOCK_LEVEL_LEASE:
-                lease_res = filter(
-                        lambda c: isinstance(c, smb2.LeaseResponse),
-                        self.create_response)[0]
+                lease_res = [c for c in self.create_response if isinstance(c, smb2.LeaseResponse)][0]
```

Comment:
implemented an iterator and a helper function to address this;

however that said, there's rarely ever more than 2 or 3 create contexts, so i think the original implementer didn't think it would be excessive to just loop through them.

Reply:
so i pushed this change (7a17714), but it actually introduced test failures, so i'm reverting it and sticking with the original logic.

i'm not comfortable enough at this point with the level of coverage in pike's included tests, so i think avoiding excessive logic changes for a futurize patch is probably preferred